src/components/Header/Header.js

```diff
@@ -95,42 +95,67 @@ const useStyles = makeStyles((theme) => ({
 const Header = (props) => {
   const classes = useStyles();
   const [anchorElement, setAnchorElement] = useState(null);
+  const [walletAnchorElement, setWalletAnchorElement] = useState(null);
+  const [walletsModalOpen, setWalletsModalOpen] = useState(true);
+  const [portis, setPortis] = useState(null);
 
-  const onConnect = async (provider) => {
-    const web3 = getWeb3(provider);
-    if (!web3.currentProvider) {
-      setWeb3Provider(provider);
-    }
+  let web3;
+
+  const onPortisClick = async () => {
+    const portisInstance = new Portis(config.wallets.portisId, 'kovan');
+
+    portisInstance.onLogin((walletAddress, email, reputation) => {
+      console.log(walletAddress, email, reputation);
+      portis.showPortis();
+    });
 
+    web3 = getWeb3(portisInstance.provider);
     const accounts = await web3.eth.getAccounts();
     props.onWalletConnect(accounts[0]); // set address in redux global state
+
+    setPortis(portisInstance);
+    setWalletsModalOpen(false); // close wallets modal
   };
 
-  const web3Modal = useWeb3Modal(onConnect);
+  const onMetaMaskClick = async () => {
+    const ethereum = window.ethereum;
+
+    const metamaskProvider = window['ethereum'];
+    web3 = getWeb3(metamaskProvider);
+
+    const accounts = await ethereum.enable();
+    props.onWalletConnect(accounts[0]); // set address in redux global state
+
+    console.log(accounts);
+    setWalletsModalOpen(false); // close wallets modal
+  };
 
-  const onWalletClick = async () => {
+  const onWalletClick = async (event) => {
     if (!props.address) {
-      // user is not connected to any wallet, open web3modal
-      web3Modal.core.connect();
+      // open wallets modal
+      setWalletsModalOpen(true);
     } else {
-      // user is already connected to wallet.
-      // TODO open menu to be able to disconnect and do other actions
+      // open menu
+      setWalletAnchorElement(event.currentTarget);
+    }
+  };
 
-      // for now, clear provider on click
-      web3Modal.core.clearCachedProvider();
-      props.onWalletConnect(null); // set address in redux global state
-      setWeb3Provider(null); // reset web3 provider
+  const onWalletDisconnet = () => {
+    if (portis) {
+      portis.logout();
+      setPortis(null);
     }
+    if (window.ethereum) {
```

Comment:
not sure there's an API for this.. is user wants to disconnect from the metamask he do this via the Metamask itself

Reply:
Yeah I figured it's going to be something like that :+1: